### PR TITLE
[RLlib] Enable RL Modules and Learner API for PPO by default

### PR DESCRIPTION
**Pull request by kouroshHakha**
Enables RL Modules and Learner API by default for PPO.
Prerequisite: https://github.com/ray-project/ray/pull/35672
Prerequisite: https://github.com/ray-project/ray/pull/35951

### CODE REVIEW

**Comment by ArturNiederfahrenhorst** [comment]
@kouroshHakha I also removed tf2 for the truncated ppo test, as it is pretty flakey right now.

File: rllib/algorithms/algorithm_config.py [hunk 1]
```diff
@@ -2507,9 +2507,8 @@ def rl_module(
         if rl_module_spec is not NotProvided:
             self.rl_module_spec = rl_module_spec
 
-        if _enable_rl_module_api is not NotProvided or self._enable_rl_module_api:
-            if not self._enable_rl_module_api:
-                self._enable_rl_module_api = _enable_rl_module_api
+        if _enable_rl_module_api is not NotProvided:
+            self._enable_rl_module_api = _enable_rl_module_api
```
Comment (ArturNiederfahrenhorst) [inline rllib/algorithms/algorithm_config.py:2511]: This reverts a change I'm skeptical about from #35180

File: rllib/tuned_examples/ppo/cartpole-truncated-ppo.py [hunk 2]
```diff
@@ -18,6 +18,7 @@
         evaluation_interval=1,
         evaluation_num_workers=1,
     )
+    .framework(eager_tracing=True)
```
Comment (ArturNiederfahrenhorst) [inline rllib/tuned_examples/ppo/cartpole-truncated-ppo.py:21]: On the new stack, tf2 is currently quite slow. Learning behaviour is the same though.
Eager tracing helps with this.

File: rllib/tuned_examples/ppo/multi_agent_pendulum_ppo.py [hunk 3]
```diff
@@ -8,17 +8,21 @@
...
-        train_batch_size=2048,
+        train_batch_size=512,
         lambda_=0.1,
-        grad_clip=0.95,
+        gamma=0.95,
         lr=0.0003,
         sgd_minibatch_size=64,
-        num_sgd_iter=10,
-        model={"fcnet_hiddens": [128, 128], "fcnet_activation": "relu"},
+        model={"fcnet_activation": "relu"},
```
Comment (ArturNiederfahrenhorst) [inline rllib/tuned_examples/ppo/multi_agent_pendulum_ppo.py:22]: The above changes reflect the HPs that we have in our single agent pendulum ppo yaml. ("pendulum-ppo.yaml")
Under the RL Modules stack, this brings a significant boost in learning behaviour.

File: rllib/tuned_examples/ppo/pendulum-ppo.yaml [hunk 4]
```diff
@@ -17,7 +17,6 @@ pendulum-ppo:
         lr: 0.0003
         sgd_minibatch_size: 64
         observation_filter: MeanStdFilter
-        seed: 42
```
Comment (ArturNiederfahrenhorst) [inline rllib/tuned_examples/ppo/pendulum-ppo.yaml:20]: This change is not directly related.
I propose it because seeding PPO and Pendulum should be 100% robust with respect to seeding.
This test would hide any missing robustness.

**Review by ArturNiederfahrenhorst — APPROVED** [review]
That was too easy 🙂

**Comment by kouroshHakha** [comment]
Nice let's goooooo!
